### PR TITLE
Remove the domain privacy protection selector in the checkout

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* as discussed we want to remove the option to select domain privacy when purchasing a new domain as it is now provided for free and people can always enable/disable the privacy protection at any time after domain purchase. We believe that we can provide privacy for all new domain registrations.

see full discussion here - p99Zz8-Ql-p2

#### Testing instructions

* add a new domain in your cart and proceed to the next step where you enter your contact information - there should be no option to enable/disable privacy

### CODE REVIEW

File: client/my-sites/checkout/checkout/domain-details-form.jsx [hunk 1]
```diff
@@ -295,20 +263,41 @@ export class DomainDetailsForm extends PureComponent {
...
-			message = this.props.translate(
-				'For your convenience, we have pre-filled your WordPress.com contact information. Please ' +
-					"review this to be sure it's the correct information you want to use for this domain."
-			);
+			if ( hasDomainProduct ) {
+				if ( hasSomeDomainsWithPrivacy ) {
+					if ( hasAllDomainsWithPrivacy ) {
+						message = this.props.translate(
+							'We have pre-filled the required contact information from your WordPress.com account. Privacy ' +
+								'Protection is included to protect your identity. {{a}}Learn more{{/a}}',
```
Comment (danhauk) [inline client/my-sites/checkout/checkout/domain-details-form.jsx]: This should be "... is included to help protect your personal information." per @kelliepeterson's comment in the thread at p99Zz8-Ql-p2

File: client/my-sites/checkout/checkout/domain-details-form.jsx [hunk 2]
```diff
@@ -295,20 +263,41 @@ export class DomainDetailsForm extends PureComponent {
...
+							{
+								components: {
+									a: <a href={ PUBLIC_VS_PRIVATE } target="_blank" rel="noopener noreferrer" />,
+								},
+							}
+						);
+					} else {
+						message = this.props.translate(
+							'We have pre-filled the required contact information from your WordPress.com account. Privacy ' +
+								'Protection is included for all eligible domains to protect your identity. {{a}}Learn more{{/a}}',
```
Comment (danhauk) [inline client/my-sites/checkout/checkout/domain-details-form.jsx]: `to protect your identity` -> `to help protect your personal information`

**Review by danhauk — CHANGES_REQUESTED** [review]
Small copy changes, but otherwise design looks good.

**Review by danhauk — APPROVED** [review]
Looks good from a product perspective. I'll let someone else approve code.

**Comment by hambai** [comment]
Uh... I have to fix the e2e tests

**Comment by daledupreez** [comment]
@hambai, if someone has more than one domain without privacy available, the copy is a bit misleading -- it reads: "Privacy Protection is unavailable for this domain."
I am not sure if it's worth splitting this up and/or pluralizing the string. (This is as much a style thing as anything else -- we could also try for a non-plural approach to the text, say "Privacy Protection is not available for any of the domains in your cart."

Other than that, this looks good.

**Review by daledupreez — COMMENTED** [review]
Rats - forgot to submit this yesterday.

File: client/lib/cart-values/cart-items.js [hunk 3]
```diff
@@ -404,20 +404,27 @@ export function hasOnlyProductsOf( cart, productSlug ) {
 /**
  * Determines whether there is at least one domain registration item in the specified shopping cart.
  *
- * @param {Object} cart - cart as `CartValue` object
+ * @param {object} cart - cart as `CartValue` object
```
[the remark this answers is not in the document]
Comment (hambai) [reply client/lib/cart-values/cart-items.js]: yep :)

**Review by daledupreez — APPROVED** [review]
LGTM!

**Pull request merged**